precompiles/gov/types.go

```diff
@@ -108,6 +133,103 @@ type TallyResultData struct {
 	NoWithVeto string
 }
 
+// NewMsgSubmitProposal creates a new MsgSubmitProposal instance.
+func NewMsgSubmitProposal(args []interface{}) (*govv1.MsgSubmitProposal, common.Address, error) {
+	if len(args) != 7 {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 7, len(args))
+	}
+
+	messages, ok := args[0].(string)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(ErrInvalidMessages, args[0])
+	}
+
+	var rawMessages []json.RawMessage
+	err := json.Unmarshal([]byte(messages), &rawMessages)
+	if err != nil {
+		return nil, common.Address{}, err
+	}
+
+	interfaceRegistry := codectypes.NewInterfaceRegistry()
+
+	authtypes.RegisterInterfaces(interfaceRegistry)
+	banktypes.RegisterInterfaces(interfaceRegistry)
+	consensustypes.RegisterInterfaces(interfaceRegistry)
+	stakingtypes.RegisterInterfaces(interfaceRegistry)
+	distrtypes.RegisterInterfaces(interfaceRegistry)
+	slashingtypes.RegisterInterfaces(interfaceRegistry)
+	govv1beta1.RegisterInterfaces(interfaceRegistry)
+	govv1.RegisterInterfaces(interfaceRegistry)
+	crisistypes.RegisterInterfaces(interfaceRegistry)
+	ibctransfertypes.RegisterInterfaces(interfaceRegistry)
+	upgradetypes.RegisterInterfaces(interfaceRegistry)
+	proposaltypes.RegisterInterfaces(interfaceRegistry)
+	cryptocodec.RegisterInterfaces(interfaceRegistry)
+
+	erc20types.RegisterInterfaces(interfaceRegistry)
+	evmtypes.RegisterInterfaces(interfaceRegistry)
+	feemarkettypes.RegisterInterfaces(interfaceRegistry)
+	inflationtypes.RegisterInterfaces(interfaceRegistry)
+
+	protoCodec := codec.NewProtoCodec(interfaceRegistry)
+
+	msgs := make([]sdk.Msg, len(rawMessages))
+	for i, message := range rawMessages {
+		var msg sdk.Msg
+		err := protoCodec.UnmarshalInterfaceJSON(message, &msg)
+		if err != nil {
+			return nil, common.Address{}, err
+		}
+
+		msgs[i] = msg
+	}
+
+	initialDeposit, ok := args[1].(*big.Int)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(ErrInvalidInitialDeposit, args[1])
+	}
+
+	proposerAddress, ok := args[2].(common.Address)
+	if !ok || proposerAddress == (common.Address{}) {
+		return nil, common.Address{}, fmt.Errorf(ErrInvalidProposer, args[2])
+	}
+
+	metadata, ok := args[3].(string)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(ErrInvalidMetadata, args[3])
+	}
+
+	title, ok := args[4].(string)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(ErrInvalidTitle, args[4])
+	}
+
+	summary, ok := args[5].(string)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(ErrInvalidSummary, args[5])
+	}
+
+	expedited, ok := args[6].(bool)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(ErrInvalidExpedited, args[6])
+	}
+
+	proposal, err := govv1.NewMsgSubmitProposal(
+		msgs,
+		sdk.Coins{sdk.NewCoin(types.BaseDenom, math.NewIntFromBigInt(initialDeposit))},
```

Comment:
Don't use a constant for the coins denom. Make sure that the denomination used is the one in the gov params (`min_deposit`)